tests/decorators/tests.py

```diff
@@ -274,6 +275,38 @@ def method(self, arg):
 
         self.assertEqual(Test().method(1), 1)
 
+    def test_method_decoration(self):
+        """
+        The method_decorator can be used to decorate a class and its methods.
+        """
+        def deco(func):
+            def _wrapper(*args, **kwargs):
+                return True
+            return _wrapper
+
+        @method_decorator(deco, name="method")
+        class Test(object):
+            def method(self):
+                return False
+
+        self.assertTrue(Test().method())
+
+    def test_invalid_non_callable_attribute_decoration(self):
+        """
+        Using the method_decorator on a non-callable attribute raises an error.
+        """
+        msg = (
+            "Cannot decorate 'prop', "
+            "as it isn't a callable attribute of "
+            "<class 'decorators.tests.MethodDecoratorTests."
+            "test_invalid_non_callable_attribute_decoration.<locals>.Test'> "
```

Comment:
Indeed. In 562ce96 I've overridden __module__ in the class used for the test, in order to standardize its `repr`.